packages/firebase_storage/firebase_storage/android/src/main/java/io/flutter/plugins/firebase/storage/TaskStateChannelStreamHandler.java

```diff
@@ -0,0 +1,114 @@
+/*
+ * Copyright 2023, the Chromium project authors.  Please see the AUTHORS file
+ * for details. All rights reserved. Use of this source code is governed by a
+ * BSD-style license that can be found in the LICENSE file.
+ */
+package io.flutter.plugins.firebase.storage;
+
+import androidx.annotation.Nullable;
+import com.google.firebase.storage.FirebaseStorage;
+import com.google.firebase.storage.StorageTask;
+import io.flutter.plugin.common.EventChannel.EventSink;
+import io.flutter.plugin.common.EventChannel.StreamHandler;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.Executor;
+import java.util.concurrent.Executors;
+
+public class TaskStateChannelStreamHandler implements StreamHandler {
+  private final FirebaseStorage androidStorage;
+  private final StorageTask androidTask;
+
+  private final String TASK_STATE_NAME = "taskState";
+  private final String TASK_APP_NAME = "appName";
+  private final String TASK_SNAPSHOT = "snapshot";
+
+  private static final Executor taskExecutor = Executors.newSingleThreadExecutor();
+  private Boolean destroyed = false;
+
+  private final Object pauseSyncObject = new Object();
+  private final Object resumeSyncObject = new Object();
+  private final Object cancelSyncObject = new Object();
+
+  public TaskStateChannelStreamHandler(FirebaseStorage androidStorage, StorageTask androidTask) {
+    this.androidStorage = androidStorage;
+    this.androidTask = androidTask;
+  }
+
+  @Override
+  public void onListen(Object arguments, EventSink events) {
+    androidTask.addOnProgressListener(
+        taskSnapshot -> {
```

Comment:
Pass down `taskExecutor` from the `FlutterFirebaseStorageTask` and put as first argument in `addOnProgressListener`, `addOnPausedListener`, `addOnSuccessListener`, `addOnCanceledListener` & `addOnFailureListener` . As it was previously: https://github.com/firebase/flutterfire/blob/master/packages/firebase_storage/firebase_storage/android/src/main/java/io/flutter/plugins/firebase/storage/FlutterFirebaseStorageTask.java#L244

Reply:
When using taskExecutor to run it will throw exception of java.lang.RuntimeException: Methods marked with @UiThread must be executed on the main thread.